### PR TITLE
Storm 2124 show requested cpu mem for each component

### PR DESCRIPTION
**Pull request by abellina**
Work done by @jerrypeng at Yahoo. I tweaked the UI a bit. Most of the change is in the generated thrift code.

Provides extra columns per component showing requested memory and cpu.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
Travis CI terminated because of exceeding the maximum time limit for jobs. 
https://travis-ci.org/apache/storm/jobs/162370376
We need to find alternative way to run Integration test.
@raghavgautam I'll disable the integration test now. Please find the way to run integration test in parallel.

**Comment by raghavgautam** [comment]
@HeartSaVioR I have started a conversation with travis-ci support, to see what we can do. I will update you on the progress.

File: storm-core/src/clj/org/apache/storm/ui/core.clj [hunk 1]
```diff
@@ -1016,6 +1019,10 @@
        "name" (.get_topology_name comp-page-info)
        "executors" (.get_num_executors comp-page-info)
        "tasks" (.get_num_tasks comp-page-info)
+       "requestedMemOnHeap" (.get (.get_resources_map comp-page-info) Config/TOPOLOGY_COMPONENT_RESOURCES_ONHEAP_MEMORY_MB)
+       "requestedMemOffHeap" (.get (.get_resources_map comp-page-info) Config/TOPOLOGY_COMPONENT_RESOURCES_OFFHEAP_MEMORY_MB)
+       "requestedCpu" (.get (.get_resources_map comp-page-info) Config/TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT)
+       "schedulerDisplayResource" (*STORM-CONF* Config/SCHEDULER_DISPLAY_RESOURCE)
```
Comment (revans2) [inline storm-core/src/clj/org/apache/storm/ui/core.clj]: Could you update the REST documentation to include the new values that are added here?

Comment (abellina) [inline storm-core/src/clj/org/apache/storm/ui/core.clj]: Sure, I'll add it.

**Comment by revans2** [comment]
My tests look good I am +1 once the docs are updated.

**Comment by jerrypeng** [comment]
Did some manual testing with the UI and everything looks good
+1

**Comment by abellina** [comment]
@revans2 I added docs and filled in docs for topology and component endpoints.

**Comment by revans2** [comment]
+1

**Comment by revans2** [comment]
I merged this into master.  The test failures are unrelated.

**Pull request merged**